pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -48,12 +50,27 @@ func init() {
 // Add creates a new DatadogAgentDeployment Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+	reconciler, err := newReconciler(mgr)
+	if err != nil {
+		return err
+	}
+	return add(mgr, reconciler)
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileDatadogAgentDeployment{client: mgr.GetClient(), scheme: mgr.GetScheme(), recorder: mgr.GetEventRecorderFor("DatadogAgentDeployment")}
+func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
+	reconciler := &ReconcileDatadogAgentDeployment{
+		client:     mgr.GetClient(),
+		scheme:     mgr.GetScheme(),
+		recorder:   mgr.GetEventRecorderFor("DatadogAgentDeployment"),
+		forwarders: datadog.NewForwardersManager(mgr.GetClient()),
+	}
+	return reconciler, mgr.Add(reconciler.forwarders)
+}
+
+type metricForwardersManager interface {
+	Register(types.NamespacedName)
+	Start(stop <-chan struct{}) error
```

Comment:
Maybe `Start()` method is not necessary in this interface

Reply:
it's necessary because we need the `reconcile.forwarders` to implement the `Runnable` interface here

```golang
mgr.Add(reconciler.forwarders)
```

Reply:
ok fair enough.
can be change by
```golang
func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
        forwarders := datadog.NewForwardersManager(mgr.GetClient())
	reconciler := &ReconcileDatadogAgentDeployment{
		client:     mgr.GetClient(),
		scheme:     mgr.GetScheme(),
		recorder:   mgr.GetEventRecorderFor("DatadogAgentDeployment"),
		forwarders: forwarders,
	}
	return reconciler, mgr.Add(forwarders)
}
```